### PR TITLE
[[ EmscriptenWindows ]] Implement windows in HTML5 using floating 'canvas' elements

### PR DESCRIPTION
**Pull request by livecodeian**
Adds windows to emscripten engine

Fixes the following bugs:
http://quality.livecode.com/show_bug.cgi?id=16254
http://quality.livecode.com/show_bug.cgi?id=16544
http://quality.livecode.com/show_bug.cgi?id=16620
http://quality.livecode.com/show_bug.cgi?id=19917

### CODE REVIEW

**Comment by livecodeali** [comment]
@livecodeian Do canvases need to be removed from the `LiveCodeDC._monitorResize.watched` list when they are destroyed? It seems like there might be an error in the `resizeCheck` function otherwise, but I might be missing something.

**Comment by livecodeian** [comment]
@livecodeali it won't cause an error as the elements will still be valid as long as there's a reference to them somewhere, however I've added removal code to clean these up as it is a resource leak.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 09dbef3

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 09dbef377c5cddf88eff16395e9c21617ef4b0b9

**Comment by livecode-vulcan** [comment]
😞 test failure 09dbef377c5cddf88eff16395e9c21617ef4b0b9

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1037)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1053)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1027)
* try-community-universal-ios-iphoneos11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.1/builds/100)
* try-community-universal-ios-iphonesimulator11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.1/builds/100)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1032)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1054)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1037)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1053)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1036)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1046)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1069)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1032)

File: engine/src/w32stack.cpp [hunk 1]
```diff
@@ -474,6 +474,11 @@ void MCStack::sethints()
 		SetWindowLongA((HWND)window->handle.window, 0, (LONG)sptr->getw()->handle.window);
 }
 
+bool MCStack::view_platform_dirtyviewonresize()
```
Comment (livecodeali) [inline engine/src/w32stack.cpp]: Looks like this is missing a `const`

**Review by livecodeali — CHANGES_REQUESTED** [review]

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 32cbd16

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 32cbd1615393e29c5dd6a340c4487fa55ad05501

**Review by livecodeali — APPROVED** [review]

**Comment by livecode-vulcan** [comment]
😎 test success 32cbd1615393e29c5dd6a340c4487fa55ad05501

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1075)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1091)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1065)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/18)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/18)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1070)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1092)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1075)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1091)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1074)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1084)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1107)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1070)

**Pull request merged**